Xamarin.Forms.Core/Internals/ImageParser.cs

```diff
@@ -0,0 +1,819 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace Xamarin.Forms.Internals
+{
+	public class GIFDecoderFormatException : Exception
```

Comment:
do you own that code, or was it borrowed from somewhere else ?

Reply:
Inspiration comes from an MIT project, but this code is a complete rewrite done by myself. The only pieces that still have some similarities are some of the internal parts of the decode and compose methods.